### ISSUE TITLE
Bug: [no-unused-vars] Used vars incorrectly reported in presence of static blocks

### ISSUE THREAD

**Issue by gebsh** [typescript-eslint/typescript-eslint#5577]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

Playground does not work

### Repro Code

```typescript
function foo() {}

export class Foo {
	static {}

	constructor() {
		foo();
	}
}
```

### ESLint Config

```javascript
module.exports = {
	parser: '@typescript-eslint/parser',
	plugins: ['@typescript-eslint/eslint-plugin'],
	rules: {
		'@typescript-eslint/no-unused-vars': 'warn',
	},
};
```

### Expected Result

No report about unused variables.

### Actual Result

The `foo` function is reported as unused even though it's used by the `Foo` class.

### Additional Info

Removing the static block from `Foo` removes the report.

Initially, I thought that this may be related to https://github.com/typescript-eslint/typescript-eslint/issues/5571, but this issue still persists after upgrading to 5.36.1.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `5.36.1` |
| `@typescript-eslint/parser`            | `5.36.1` |
| `TypeScript`                           | `4.8.2` |
| `ESLint`                               | `8.23.0` |
| `node`                                 | `18.8.0` |

**Comment by jcurtis** [comment]
Reverting to 5.36.0 fixes this for me. I believe 5.36.1 introduced a compatibility issue with typescript 4.8 with #5574
